tap_db2/__init__.py

```diff
@@ -56,83 +55,134 @@
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+# Define data types
+
+# Full list
+#BIGINT - i
+#BLOB - ignore for now
+#CHARACTER - s
+#CLOB - ignore for now
+#DATE - d
+#DECIMAL - f
+#DOUBLE - f
+#INTEGER - i
+#SMALLINT - i
+#TIMESTAMP - d
+#VARCHAR - s
+#XML - s
+
 STRING_TYPES = set(
     [
-        "binary",
         "char",
-        "enum",
-        "longtext",
-        "mediumtext",
-        "nchar",
-        "nvarchar",
-        "text",
-        "uniqueidentifier",
-        "varbinary",
+        "character",
         "varchar",
+        "xml",
     ]
 )
 
 BYTES_FOR_INTEGER_TYPE = {
-    "tinyint": 1,
     "smallint": 2,
-    "mediumint": 3,
-    "int": 4,
     "integer": 4,
-    "real": 4,
+    "int": 4,
     "bigint": 8,
 }
 
-FLOAT_TYPES = set(
+# REAL -37
+# DOUBLE -307
+# DECFLOAT(16/8) -383
+# DECFLOAT(34/16) -6143
+# For practicality this will be limited to -50
+MAX_EXPONENT=-50
+
+FLOAT_TYPE_EXPONENT = {
+  "real":-37,
+  "double":MAX_EXPONENT,
+  "decfloat":MAX_EXPONENT,
+}
+
+DECIMAL_TYPES = set(
     [
-        "decfloat",
-        "double",
-        "float",
-        "money",
+        "decimal",
+        "numeric"
     ]
 )
 
 DATETIME_TYPES = set(
     [
-        "date",
-        "datetime",
-        "datetime2",
-        "smalldatetime",
-        "time",
         "timestamp",
     ]
 )
 
-VARIANT_TYPES = set(["json"])
+DATE_TYPES = set(
+        [
+            "date",
+        ]
+)
 
+def default_date_format():
+    return False
 
-def schema_for_column(c):
+def default_offset_value():
+    """
+    Function included to remain consistent with MSSQL tap using a False-returning function
+    for default_date_format
+    """
+    return False
+  
+def default_singer_decimal():
+    """
+    singer_decimal can be enabled in the the config, which will use singer.decimal as a format and string as the type
+    use this for large/precise numbers
+    """
+    return False
+
+def schema_for_column(c,config):
     """Returns the Schema object for the given Column."""
     data_type = c.data_type.strip().lower()
 
     inclusion = "available"
 
+    use_date_data_type_format = config.get('use_date_datatype') or default_date_format()
+    use_singer_decimal = config.get('use_singer_decimal') or default_singer_decimal()
+
     if c.is_primary_key == 1:
         inclusion = "automatic"
 
     result = Schema(inclusion=inclusion)
+    
+    # the tap-oracle behaviour:
+    # integer (small/big/normal) to integer - sysmaxsize can hold all these
+    # all decimals, floats and numerics to number/singer.decimal
+    # number with no c.numeric_scale to integer
 
-    if data_type == "bit":
```

Comment:
I'm wondering if we should support the bit datatype. We have in other taps?

Reply:
I've added back boolean and bit. However, this page shows it's unlikely to ever come up:

https://www.ibm.com/docs/en/db2/10.5?topic=type-restrictions-boolean-data

Boolean can't be used as a column data type and 'bit' is invalid when used in SQL. Perhaps it will pop up in DB2 Z/OS rather than LUW so it's good to have it just in case.